Assets/_BForBoss/_Core/Scripts/Debug/DebugWindow.cs

```diff
@@ -135,25 +137,53 @@ private void OpenPanel()
         {
             _currentState = _stateManager.GetState();
             _stateManager.SetState(State.Pause);
-            IInputSettings input = FindObjectOfType<FirstPersonPlayer>();
-            input.SwapToUIActions();
+            _freezeActionsUtility.LockInput();
```

Comment:
I see, this looks like this was an issue even before this fix came into place.Like if you press escape with the debugWindow open it will reset you back to starting position and let you move

Reply:
So current implementation is that when the DebugWindow, the game is Paused, so when you Press the Escape Button, the game will go into PlayMode. I think trying to  Pause/Resume the game while the DebugWindow is still active seems a bit redundant to be honest So either 

1) We don't accept the Escape Button to register while the DebugWindow is open (which might be a bit tedious)
2) Or we create a new State, i.e. `State.Debug`, and set the game to that when the DebugWindow is open, so that Escape Key has no bearing while the Debug Window is open (I personally like this option the best!!)

Reply:
What do you think @m4rtint ?

Reply:
So one Line 139: `_stateManager.SetState(State.Debug);`

Reply:
I think creating a new input action map would be best. so we get to control what is allowed in that input state.

Reply:
As discussed, the action map will remain hardcoded:
The [Spike](https://perigongames.atlassian.net/browse/BFB-348?atlOrigin=eyJpIjoiMjI3NDk2NWE1YzkwNGRhZWFiZWI1YjRkMjQ3MTRiNTYiLCJwIjoiaiJ9) to talk about which entity should/should not be time dependent: